geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteRegionFunctionOp.java

```diff
@@ -108,11 +111,19 @@ private static void execute(ExecutablePool pool, String region, String function,
 
         isReexecute = true;
         resultCollector.clearResults();
+        failedNodes = allocateFailedNodesIfEmpty(failedNodes);
         failedNodes.clear();
       }
     } while (true);
   }
 
+  private static Set<String> allocateFailedNodesIfEmpty(Set<String> failedNodes) {
+    if (failedNodes.equals(Collections.emptySet())) {
+      return new HashSet();
```

Comment:
ooh and this needs to be `new HashSet<>()` to eliminate the unchecked assignment warning

Reply:
Yep. Thanks!